arroyo/processing/processor.py

```diff
@@ -79,21 +79,24 @@ def _create_strategy() -> None:
 
         def on_partitions_assigned(partitions: Mapping[Partition, int]) -> None:
             logger.info("New partitions assigned: %r", partitions)
```

Comment:
Shouldn't this go inside the if partitions check as well?

Reply:
I think it's useful to know when the partition assigned callback gets called even if it's empty

Reply:
Ok. I was under the impression that partitions could be optional. In that case, it would cause an exception.